src/ising.jl

```diff
@@ -36,19 +49,21 @@ function energy(σ::Union{Vector, NTuple}, ig::MetaGraph)
 
     # linear
     for i ∈ vertices(ig)
-        h = get_prop(ig, i, :h)   
-        energy += h * σ[i]     
+        h = get_prop(ig, i, :h)  
+        energy += h * σ[i]
     end    
     -energy
 end
     
 """
 Create a graph that represents the Ising Hamiltonian.
 """
-function ising_graph(instance::String, L::Int, β::Number=1)
+function ising_graph(instance::Union{String, Dict}, L::Int, β::Number=1)
 
     # load the Ising instance
-    ising = CSV.File(instance, types=[Int, Int, Float64])
+    if typeof(instance) == String 
```

Comment:
What happens if instace is a `Dict`?

Reply:
Ideally, the same. One way or another we should have the ability to load instances from Dict. This is extremely useful for testing.

Reply:
Yes, but now we only deal with the case when `instance` is an instance of `String`. What should `ising` be when `instance` is a `Dict`?

Reply:
I will remove it for now. Maybe we should write 2 different functions unless of course there is a clean way to do it in one.
The ideal is that, in the case of a Dict, ising should allow somehow iterate keys and values